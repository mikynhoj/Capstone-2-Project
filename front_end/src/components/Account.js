import React, { useEffect, useState, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useParams } from "react-router-dom";
import { getTransactions, getAccounts } from "../helpers/actionCreators";
import Transaction from "./Transaction";
import FilterTransactionsForm from "./FilterTransactionsForm";
import ReactPaginate from "react-paginate";
import "../styles/Transactions.css";
import { bool, node } from "prop-types";
import { useTransition, animated } from "react-spring";
import styled from "styled-components";
import "../styles/Account.css";

//Toggle Form
const Inner = styled.div`
  &:before,
  &:after {
    content: "";
    display: table;
  }
`;

const visibleStyle = { height: "auto", opacity: 1, overflow: "visible" };
const hiddenStyle = { opacity: 0, height: 0, overflow: "hidden" };

function getElementHeight(ref) {
  return ref.current ? ref.current.getBoundingClientRect().height : 0;
}

/** The children of this component will slide down on mount and will slide up on unmount */
const SlideToggleContent = ({ isVisible, children, forceSlideIn }) => {
  const isVisibleOnMount = useRef(isVisible && !forceSlideIn);
  const containerRef = useRef(null);
  const innerRef = useRef(null);

  const transitions = useTransition(isVisible, null, {
    enter: () => async (next, cancel) => {
      const height = getElementHeight(innerRef);

      cancel();

      await next({ height, opacity: 1, overflow: "hidden" });
      await next(visibleStyle);
    },
    leave: () => async (next, cancel) => {
      const height = getElementHeight(containerRef);

      cancel();

      await next({ height, overflow: "hidden" });
      await next(hiddenStyle);

      isVisibleOnMount.current = false;
    },
    from: isVisibleOnMount.current ? visibleStyle : hiddenStyle,
    unique: true,
  });

  return transitions.map(({ item: show, props: springProps, key }) => {
    if (show) {
      return (
        <animated.div ref={containerRef} key={key} style={springProps}>
          <Inner ref={innerRef}>{children}</Inner>
        </animated.div>
      );
    }

    return null;
  });
};

SlideToggleContent.defaultProps = {
  forceSlideIn: false,
};

SlideToggleContent.propTypes = {
  /** Should the component mount it's childeren and slide down */
  isVisible: bool.isRequired,
  /** Makes sure the component always slides in on mount. Otherwise it will be immediately visible if isVisible is true on mount */
  forceSlideIn: bool,
  /** The slidable content elements */
  children: node.isRequired,
};

const Account = ({ test }) => {
  // Transaction info
  // const { account_id } = test || useParams();
  const { account_id } = useParams();
  const dispatch = useDispatch();
  const accounts = useSelector((state) => state.accounts);
  const accountSpecificTransactions = useSelector(
    (state) => state.accountTransactions[`${account_id}`]
  );
  const item_id = useSelector((state) => state.item_id);
  const institution_id = useSelector((state) => state.institution_id);
  const token = useSelector((state) => state.token);
  const _token = useSelector((state) => state._token);
  const access_token = useSelector((state) => state.access_token);
  const logged_in = useSelector((state) => state.logged_in);
  const [clearedOnce, setClearedOnce] = useState(false);
  const [account, setAccount] = useState();
  useEffect(() => {
    if (!clearedOnce && accountSpecificTransactions) {
      setClearedOnce(true);
      dispatch({ type: "RESET_ACCOUNT_TRANSACTIONS" });
    } else if (
      !accountSpecificTransactions &&
      logged_in &&
      _token &&
      access_token
    ) {
      dispatch({
        type: "SET_CURRENT_LOCATION",
        currentLocation: window.location.pathname,
      });
      dispatch(getTransactions(_token, access_token, account_id));
    }
    if (
      !accounts &&
      _token &&
      access_token &&
      item_id &&
      institution_id &&
      token
    ) {
      dispatch(getAccounts(_token, access_token, false));
    }
    if (accounts) {
      setAccount(
        (account) =>
          accounts.filter((account) => account.account_id === account_id)[0]
      );
    }
  });
  ///Pagination
  const [currentPage, setCurrentPage] = useState(0);
  const PER_PAGE = 25;
  const offset = currentPage * PER_PAGE;
  let currentTransactions;
  let pageCount;
  if (accountSpecificTransactions) {
    currentTransactions = accountSpecificTransactions
      .slice(offset, offset + PER_PAGE)
      .map((transaction, index) => {
        return <Transaction key={index} transaction={transaction} />;
      });
    pageCount = Math.ceil(accountSpecificTransactions.length / PER_PAGE);
  }
  function handlePageClick({ selected: selectedPage }) {
    setCurrentPage(selectedPage);
  }

  // ToggleForm
  const [isVisible, setIsVisible] = useState(false);
  return (
    <div className="Account">
      <h4 className="Account-title">
        Transactions for Account ...{account && account.mask}
      </h4>
      <button
        type="button"
        className="btn btn-secondary Account-filter-button"
        onClick={() => setIsVisible(!isVisible)}
      >
        {isVisible ? "Close filters" : "Filter Transactions"}
      </button>
      <SlideToggleContent isVisible={isVisible}>
        <>
          <FilterTransactionsForm setCurrentPage={setCurrentPage} />
        </>
      </SlideToggleContent>
      <div className="Transactions">
        {accountSpecificTransactions && (
          <>
            {accountSpecificTransactions.length === 0 && (
              <div>
                <h5>
                  There are no transactions for this account or filter request
                </h5>
              </div>
            )}
            {currentTransactions}

            <ReactPaginate
              previousLabel={"← Previous"}
              nextLabel={"Next →"}
              pageCount={pageCount}
              onPageChange={handlePageClick}
              containerClassName={`pagination ${
                accountSpecificTransactions.length === 0 ? "d-none" : ""
              }`}
              previousLinkClassName={"pagination__link"}
              nextLinkClassName={"pagination__link"}
              disabledClassName={"pagination__link--disabled"}
              activeClassName={"pagination__link--active"}
            />
          </>
        )}
      </div>
    </div>
  );
};

export default Account;
