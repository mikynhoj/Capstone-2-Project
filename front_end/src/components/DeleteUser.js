import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Button } from "reactstrap";
import { deleteProfile, getAccounts } from "../helpers/actionCreators";
import "../styles/DeleteUser.css";
const DeleteUser = () => {
  const dispatch = useDispatch();
  dispatch({
    type: "SET_CURRENT_LOCATION",
    currentLocation: window.location.pathname,
  });
  const INITIAL_STATE = {
    password: "",
  };
  const token = useSelector((state) => state.token);
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({ ...data, [name]: value }));
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      dispatch(deleteProfile(formData, token));
      setFormData(INITIAL_STATE);
    } catch (error) {
      setFormData(INITIAL_STATE);
      alert(error[0]);
    }
  };

  const _token = useSelector((state) => state._token);
  const access_token = useSelector((state) => state.access_token);
  const item_id = useSelector((state) => state.item_id);
  const institution_id = useSelector((state) => state.institution_id);
  const accounts = useSelector((state) => state.accounts);
  useEffect(() => {
    if (
      !accounts &&
      _token &&
      access_token &&
      item_id &&
      institution_id &&
      token
    ) {
      dispatch(getAccounts(_token, access_token, false));
    }
  });
  return (
    <div className="Delete-user">
      <div style={{ marginTop: "15px" }}>
        We regret that you would like to delete your account. Please enter your
        password below in order to do so.
      </div>
      <form onSubmit={handleSubmit}>
        <div className="Delete-user-items form-group">
          <label htmlFor="password">
            <b>Password</b>
          </label>
          <div className="Input-password">
            <input
              autoComplete="new-password"
              type="password"
              required={true}
              id="password"
              name="password"
              className="form-control"
              onChange={handleChange}
              value={formData.password}
            />
          </div>
        </div>

        <Button color="danger">Submit</Button>
      </form>
    </div>
  );
};

export default DeleteUser;
