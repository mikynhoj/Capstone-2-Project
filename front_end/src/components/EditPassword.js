import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Button } from "reactstrap";
import { editPassword, getAccounts } from "../helpers/actionCreators";
import "../styles/EditUsername.css";

const EditUsername = () => {
  const dispatch = useDispatch();
  dispatch({
    type: "SET_CURRENT_LOCATION",
    currentLocation: window.location.pathname,
  });
  const INITIAL_STATE = {
    password: "",
    new_password: "",
    new_password_copy: "",
  };
  const token = useSelector((state) => state.token);
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [showPassword1, setShowPassword1] = useState(false);
  const [showPassword2, setShowPassword2] = useState(false);
  const toggleShowPassword1 = () => {
    setShowPassword1((password) => !password);
  };
  const toggleShowPassword2 = () => {
    setShowPassword2((password) => !password);
  };
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({ ...data, [name]: value }));
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      if (formData.new_password !== formData.new_password_copy) {
        alert("New Passwords do not match");
        return;
      }
      dispatch(editPassword(formData, token));
    } catch (error) {
      setFormData(INITIAL_STATE);
      alert(error[0]);
    }
  };

  const _token = useSelector((state) => state._token);
  const access_token = useSelector((state) => state.access_token);
  const item_id = useSelector((state) => state.item_id);
  const institution_id = useSelector((state) => state.institution_id);
  const accounts = useSelector((state) => state.accounts);
  useEffect(() => {
    if (
      !accounts &&
      _token &&
      access_token &&
      item_id &&
      institution_id &&
      token
    ) {
      dispatch(getAccounts(_token, access_token, false));
    }
  });
  return (
    <div className="Edit-user">
      <h3 style={{ marginTop: "15px" }}>Edit Username</h3>
      <form onSubmit={handleSubmit}>
        <div className="Edit-user-items form-group">
          <label htmlFor="password">
            <b>Current Password</b>
          </label>
          <div className="Input-password">
            <input
              autoComplete="new-password"
              type="password"
              required={true}
              id="password"
              name="password"
              className="form-control"
              onChange={handleChange}
              value={formData.password}
            />
          </div>
        </div>
        <div className="Edit-user-items form-group">
          <label htmlFor="new_password">
            <b>New Password</b>
          </label>
          <div className="Input-password">
            <input
              type={showPassword1 ? "text" : "password"}
              required={true}
              id="new_password"
              name="new_password"
              className="form-control"
              onChange={handleChange}
              value={formData.new_password}
            />
            <Button
              type="button"
              color="primary"
              className="password-button"
              onClick={() => toggleShowPassword1()}
            >
              {showPassword2 ? "Hide Password" : "Show Password"}
            </Button>
          </div>
        </div>
        <div className="Edit-user-items form-group">
          <label htmlFor="new_password_copy">
            <b>Re-type New Password</b>
          </label>
          <div className="Input-password">
            <input
              type={showPassword2 ? "text" : "password"}
              required={true}
              id="new_password_copy"
              name="new_password_copy"
              className="form-control"
              onChange={handleChange}
              value={formData.new_password_copy}
            />
            <Button
              type="button"
              color="primary"
              className="password-button"
              onClick={() => toggleShowPassword2()}
            >
              {showPassword2 ? "Hide Password" : "Show Password"}
            </Button>
          </div>
        </div>
        <Button color="primary" className="Login-submit">
          Submit
        </Button>
      </form>
    </div>
  );
};

export default EditUsername;
