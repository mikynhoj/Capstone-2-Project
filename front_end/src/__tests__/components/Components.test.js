import React from "react";
import { render, fireEvent } from "../test-utils";
import Login from "../../components/Login";
import Signup from "../../components/Signup";
import Account from "../../components/Account";
import "@testing-library/jest-dom";
import AccountTrends from "../../components/AccountsTrends";
import DeleteUser from "../../components/DeleteUser";
import EditPassword from "../../components/EditPassword";
import EditUsername from "../../components/EditUsername";
import FilterTransactionsForm from "../../components/FilterTransactionsForm";
import FilterAllTransactionsForm from "../../components/FilterAllTransactionsForm";
import IdleModal from "../../components/IdleModal";
import LinkLoggedIn from "../../components/LinkLoggedIn";
import NavBar from "../../components/NavBar";
import { BrowserRouter } from "react-router-dom";
// Component to be tested

describe("<Login />", () => {
  describe("render()", () => {
    test("renders the login component", () => {
      const { asFragment } = render(<Login />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe("<Signup />", () => {
  describe("render()", () => {
    test("renders the Signup component", () => {
      const { asFragment } = render(<Signup />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe("<Login /> And <Signup/>", () => {
  describe("onClick()", () => {
    test("successfully calls the onClick handler and toggles between the two components", () => {
      const { getAllByText, getByText } = render(<Login />);
      const signupButton = getByText("Signup");
      const titles = getAllByText("Login");
      expect(titles[1]).toBeInTheDocument();
      fireEvent.click(signupButton);
      expect(titles[1]).not.toBeInTheDocument();
    });
  });
});
describe("<Account />", () => {
  describe("render()", () => {
    test("renders the Account component", () => {
      const { asFragment } = render(<Account test={true} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<AccountTrends />", () => {
  describe("render()", () => {
    test("renders the AccountTrends component", () => {
      const { asFragment } = render(<AccountTrends />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe("<DeleteUser />", () => {
  describe("render()", () => {
    test("renders the Delete User component", () => {
      const { asFragment } = render(<DeleteUser />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<EditPassword />", () => {
  describe("render()", () => {
    test("renders the Edit Password component", () => {
      const { asFragment } = render(<EditPassword />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<EditUsername />", () => {
  describe("render()", () => {
    test("renders the Edit Username component", () => {
      const { asFragment } = render(<EditUsername />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<FilterTransactions />", () => {
  describe("render()", () => {
    test("renders the Filter Transactions component", () => {
      const { asFragment } = render(<FilterTransactionsForm test={true} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<FilterAllTransactions />", () => {
  describe("render()", () => {
    test("renders the Filter All Transactions component", () => {
      const { asFragment } = render(<FilterAllTransactionsForm />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<IdleModal />", () => {
  describe("render()", () => {
    test("renders the IdleModal component", () => {
      const { asFragment } = render(<IdleModal />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<LinkLoggedIn />", () => {
  describe("render()", () => {
    test("renders the LinkLoggedIn component", () => {
      const { asFragment } = render(<LinkLoggedIn />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
describe("<Navbar />", () => {
  describe("render()", () => {
    test("renders the Navbar component", () => {
      const { asFragment } = render(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
