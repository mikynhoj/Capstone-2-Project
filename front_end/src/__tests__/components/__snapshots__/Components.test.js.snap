// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> render() renders the Account component 1`] = `
<DocumentFragment>
  <div
    class="Account"
  >
    <h4
      class="Account-title"
    >
      Transactions for Account ...
    </h4>
    <button
      class="btn btn-secondary Account-filter-button"
      type="button"
    >
      Filter Transactions
    </button>
    <div
      class="Transactions"
    />
  </div>
</DocumentFragment>
`;

exports[`<AccountTrends /> render() renders the AccountTrends component 1`] = `
<DocumentFragment>
  <div
    class="AccountTrends"
  >
    <h4
      class="AccountTrends-title"
    >
      Transactions for Multiple Accounts
    </h4>
    <button
      class="btn btn-secondary AccountTrends-filter-button"
      type="button"
    >
      Close Filters
    </button>
    <div>
      ...loading
    </div>
    <div
      style="height: auto; opacity: 1; overflow: visible;"
    >
      <div
        class="sc-gsTCUz epgQsa"
      />
    </div>
    <div
      class="Transactions"
    />
  </div>
</DocumentFragment>
`;

exports[`<DeleteUser /> render() renders the Delete User component 1`] = `
<DocumentFragment>
  <div
    class="Delete-user"
  >
    <div
      style="margin-top: 15px;"
    >
      We regret that you would like to delete your account. Please enter your password below in order to do so.
    </div>
    <form>
      <div
        class="Delete-user-items form-group"
      >
        <label
          for="password"
        >
          <b>
            Password
          </b>
        </label>
        <div
          class="Input-password"
        >
          <input
            autocomplete="new-password"
            class="form-control"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        class="btn btn-danger"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`<EditPassword /> render() renders the Edit Password component 1`] = `
<DocumentFragment>
  <div
    class="Edit-user"
  >
    <h3
      style="margin-top: 15px;"
    >
      Edit Username
    </h3>
    <form>
      <div
        class="Edit-user-items form-group"
      >
        <label
          for="password"
        >
          <b>
            Current Password
          </b>
        </label>
        <div
          class="Input-password"
        >
          <input
            autocomplete="new-password"
            class="form-control"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="Edit-user-items form-group"
      >
        <label
          for="new_password"
        >
          <b>
            New Password
          </b>
        </label>
        <div
          class="Input-password"
        >
          <input
            class="form-control"
            id="new_password"
            name="new_password"
            required=""
            type="password"
            value=""
          />
          <button
            class="password-button btn btn-primary"
            type="button"
          >
            Show Password
          </button>
        </div>
      </div>
      <div
        class="Edit-user-items form-group"
      >
        <label
          for="new_password_copy"
        >
          <b>
            Re-type New Password
          </b>
        </label>
        <div
          class="Input-password"
        >
          <input
            class="form-control"
            id="new_password_copy"
            name="new_password_copy"
            required=""
            type="password"
            value=""
          />
          <button
            class="password-button btn btn-primary"
            type="button"
          >
            Show Password
          </button>
        </div>
      </div>
      <button
        class="Login-submit btn btn-primary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`<EditUsername /> render() renders the Edit Username component 1`] = `
<DocumentFragment>
  <div
    class="Edit-user"
  >
    <h3
      style="margin-top: 15px;"
    >
      Edit Username
    </h3>
    <form>
      <div
        class="Edit-user-items form-group"
      >
        <label
          for="username"
        >
          <b>
            New Username
          </b>
        </label>
        <input
          class="form-control"
          id="new_username"
          name="new_username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="Edit-user-items form-group"
      >
        <label
          for="password"
        >
          <b>
            Password
          </b>
        </label>
        <div
          class="Input-password"
        >
          <input
            autocomplete="new-password"
            class="form-control"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        class="Login-submit btn btn-primary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`<FilterAllTransactions /> render() renders the Filter All Transactions component 1`] = `
<DocumentFragment>
  <form
    class="FilterAllTransactionsForm"
  >
    <div
      class="FilterAllTransactions"
    >
      <div
        class="FilterAllTransactions-top"
      >
        <div
          id="FilterAllTransactions-top-left"
        >
          <label
            for="accountMask"
          >
            Select Time Period:
          </label>
          <div
            class=" css-2b097c-container"
          >
            <div
              class="select__control css-yk16xz-control"
            >
              <div
                class="select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="select__placeholder css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-3-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class="select__indicator-separator css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="time_period"
              type="hidden"
              value=""
            />
          </div>
          <input
            autocomplete="off"
            required=""
            style="opacity: 0; height: 0px;"
            tabindex="-1"
            value=""
          />
        </div>
        <div
          id="FilterAllTransactions-top-right"
        >
          <label
            for="accountMask"
            id="accountMask-label"
          >
            Select an Account: (Default selects all accounts)
          </label>
          <div
            class="basic-multi-select css-2b097c-container"
          >
            <div
              class="select__control css-yk16xz-control"
            >
              <div
                class="select__value-container select__value-container--is-multi css-g1d714-ValueContainer"
              >
                <div
                  class="select__placeholder css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-4-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class="select__indicator-separator css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="account_ids"
                type="hidden"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="FilterAllTransactions-bottom"
      >
        <div
          class="FilterAllTransactions-bottom-label-input"
        >
          <label
            for="amount_floor"
          >
            Lowest Amount: 
          </label>
          <input
            class="form-control"
            id="amount_floor"
            name="amount_floor"
            step=".01"
            type="number"
            value=""
          />
        </div>
        <div
          class="FilterAllTransactions-bottom-label-input"
        >
          <label
            for="amount_ceiling"
          >
            Highest Amount: 
          </label>
          <input
            class="form-control"
            id="amount_ceiling"
            name="amount_ceiling"
            step=".01"
            type="number"
            value=""
          />
        </div>
      </div>
      <div>
        <button
          class="FilterAllTransactions-submit btn btn-primary"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<FilterTransactions /> render() renders the Filter Transactions component 1`] = `
<DocumentFragment>
  <form
    class="FilterTransactionsForm"
  >
    <div
      class="FilterTransactions"
    >
      <div
        class="FilterTransactions-top"
      >
        <div>
          <label
            for="name"
          >
            Transaction Name:
          </label>
          <input
            class="form-control"
            id="name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="start_date"
          >
            Between Dates (Begin): 
          </label>
          <input
            class="form-control"
            id="start_date"
            name="start_date"
            type="date"
            value=""
          />
        </div>
        <div>
          <label
            for="end_date"
          >
            Between Dates (End): 
          </label>
          <input
            class="form-control"
            id="end_date"
            name="end_date"
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="FilterTransactions-bottom"
      >
        <div>
          <label
            class="FilterTransactions-bottom-labels"
            for="categories"
          >
            Categories
          </label>
          <div
            class=" css-2b097c-container"
            id="categories"
          >
            <div
              class="select__control css-yk16xz-control"
            >
              <div
                class="select__value-container select__value-container--is-multi css-g1d714-ValueContainer"
              >
                <div
                  class="select__placeholder css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class="select__indicator-separator css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="categories"
                type="hidden"
              />
            </div>
          </div>
        </div>
        <div>
          <label
            class="FilterTransactions-bottom-labels"
            for="amount_floor"
          >
            Lowest Amount:
          </label>
          <input
            class="form-control"
            id="amount_floor"
            name="amount_floor"
            step=".01"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            class="FilterTransactions-bottom-labels"
            for="amount_ceiling"
          >
            Highest Amount: 
          </label>
          <input
            class="form-control"
            id="amount_ceiling"
            name="amount_ceiling"
            step=".01"
            type="number"
            value=""
          />
        </div>
      </div>
      <div>
        <button
          class="FilterTransactions-submit btn btn-primary"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<IdleModal /> render() renders the IdleModal component 1`] = `<DocumentFragment />`;

exports[`<LinkLoggedIn /> render() renders the LinkLoggedIn component 1`] = `
<DocumentFragment>
  <div
    class="Account-Summaries"
  >
    <h1
      class="display-4"
    >
      Accounts
    </h1>
    <h6>
      ...loading
    </h6>
    <div
      class="Account-Summaries-box"
    />
  </div>
</DocumentFragment>
`;

exports[`<Login /> render() renders the login component 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div>
      <button
        class="btn btn-primary active"
      >
        Login
      </button>
      <button
        class="btn btn-secondary "
      >
        Signup
      </button>
    </div>
    <div
      class="card"
    >
      <h4
        id="title"
      >
        Login
      </h4>
      <div
        class="card-body"
      >
        <form
          class="Login-form"
        >
          <div
            class="Login-form-items form-group"
          >
            <label
              for="username"
            >
              <b>
                Username
              </b>
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="Login-form-items form-group"
          >
            <label
              for="password"
            >
              <b>
                Password
              </b>
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <button
            class="Login-submit btn btn-primary"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Navbar /> render() renders the Navbar component 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="bg-primary navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Cash Counselor
      </a>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`<Signup /> render() renders the Signup component 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <h4
      id="title"
    >
      Signup
    </h4>
    <div
      class="card-body"
    >
      <form>
        <div
          class="Signup-form-items form-group"
        >
          <label
            for="username"
          >
            <b>
              Username
            </b>
          </label>
          <input
            class="form-control"
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="Signup-form-items form-group"
        >
          <label
            for="first_name"
          >
            <b>
              First Name
            </b>
          </label>
          <input
            class="form-control"
            id="first_name"
            name="first_name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="Signup-form-items form-group"
        >
          <label
            for="password"
          >
            <b>
              Password
            </b>
          </label>
          <div
            class="Input-password"
          >
            <input
              autocomplete="new-password"
              class="form-control Signup-password"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <button
              class="password-button btn btn-primary"
              type="button"
            >
              Show Password
            </button>
          </div>
        </div>
        <div
          class="Signup-form-items form-group"
        >
          <label
            for="password_copy"
          >
            <b>
              Re-type Password
            </b>
          </label>
          <div
            class="Input-password"
          >
            <input
              class="form-control  Signup-password"
              id="password_copy"
              name="password_copy"
              required=""
              type="password"
              value=""
            />
            <button
              class="password-button btn btn-primary"
              type="button"
            >
              Show Password
            </button>
          </div>
        </div>
        <button
          class="Login-submit btn btn-primary"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
